jenkins/Hono-Release-Pipeline.groovy

```diff
@@ -0,0 +1,144 @@
+                    defaultValue: '',
+                    description: 'The credentials to use during checkout from git',
+                    name: 'CREDENTIALS_ID',
+                    required: true)])])
+    try {
+        checkOut()
+        setReleaseVersionAndBuild()
+        commitAndTag()
+        setNextVersion()
+        commitAndPush()
+        copyArtifacts()
+        utils.publishJavaDoc('target/site/apidocs', true)
+        currentBuild.result = 'SUCCESS'
+    } catch (err) {
+        currentBuild.result = 'FAILURE'
+        echo "Error: ${err}"
+    }
+    finally {
+        echo "Build status: ${currentBuild.result}"
+        utils.notifyBuildStatus()
+    }
+}
+
+/**
+ * Checks out the specified branch from git repo
+ *
+ */
+def checkOut() {
+    stage('Checkout') {
+        echo "Check out branch: origin/${params.BRANCH}"
+        checkout([$class                           : 'GitSCM',
+                  branches                         : [[name: "origin/${params.BRANCH}"]],
+                  doGenerateSubmoduleConfigurations: false,
+                  extensions                       : [[$class: 'WipeWorkspace'],
+                                                      [$class: 'LocalBranch']],
+                  userRemoteConfigs                : [[credentialsId: "${params.CREDENTIALS_ID}", url: 'ssh://github.com/eclipse/hono.git']]])
+    }
+}
+
+/**
+ * Set version to RELEASE_VERSION and build using maven
+ *
+ */
+def setReleaseVersionAndBuild() {
+    stage('Build') {
+        withMaven(maven: 'apache-maven-latest', jdk: 'jdk1.8.0-latest', options: [jacocoPublisher(disabled: true), artifactsPublisher(disabled: true)]) {
+            sh "mvn versions:set -DgenerateBackupPoms=false -DnewVersion=${params.RELEASE_VERSION}"
+            sh 'mvn clean install javadoc:aggregate -Dmaven.test.failure.ignore=false -DenableEclipseJarSigner=true -DsnapshotDependencyAllowed=false -Ddocker.skip.build=true'
+        }
+    }
+}
+
+/**
+ * Commit and tag to RELEASE_VERSION
+ *
+ */
+def commitAndTag() {
+    stage("Commit and tag release version") {
+        sh ''' 
+            git add pom.xml \\*/pom.xml
+            git commit -m "Release ${RELEASE_VERSION}"
+            git tag ${RELEASE_VERSION}
+           '''
+    }
+
+}
+
+/**
+ * Set version to NEXT_VERSION using maven
+ *
+ */
+def setNextVersion() {
+    stage("Set next version") {
+        withMaven(maven: 'apache-maven-latest', jdk: 'jdk1.8.0-latest', options: [jacocoPublisher(disabled: true), artifactsPublisher(disabled: true)]) {
+            sh "mvn versions:set -DallowSnapshots=true -DgenerateBackupPoms=false -DnewVersion=${NEXT_VERSION}"
+        }
+    }
+}
+
+/**
+ * Commit and push to the Hono repo
+ *
+ */
+def commitAndPush() {
+    stage("Commit and push") {
+        sh ''' 
+            git add pom.xml \\*/pom.xml
+            git commit -m "Bump version to ${NEXT_VERSION}"
```

Comment:
is there a reason, why the parameters are sometimes referred to as `${params.NEXT_VERSION}` and sometimes as just `${NEXT_VERSION}`?

Reply:
Whenever I used sh command to execute shell scripts, I directly invoked the parameter without  prefix `params`.  The parameters are exposed as environment variables of the same name into shell.